pennylane/templates/state_preparations/mottonen.py

```diff
@@ -14,83 +14,107 @@
 r"""
 Contains the MottonenStatePreparation template.
 """
+from typing import Optional
+
 import numpy as np
 
 import pennylane as qml
 from pennylane.operation import Operation
+from pennylane.typing import TensorLike
 
 
-# pylint: disable=len-as-condition,arguments-out-of-order,consider-using-enumerate
 def gray_code(rank):
-    """Generates the Gray code of given rank.
+    """Generates the Gray code of given rank, as numeric output.
 
     Args:
         rank (int): rank of the Gray code (i.e. number of bits)
-    """
-
-    def gray_code_recurse(g, rank):
-        k = len(g)
-        if rank <= 0:
-            return
-
-        for i in range(k - 1, -1, -1):
-            char = "1" + g[i]
-            g.append(char)
-        for i in range(k - 1, -1, -1):
-            g[i] = "0" + g[i]
-
-        gray_code_recurse(g, rank - 1)
-
-    g = ["0", "1"]
-    gray_code_recurse(g, rank - 1)
-
-    return g
-
-
-def _matrix_M_entry(row, col):
-    """Returns one entry for the matrix that maps alpha to theta.
-
-    See Eq. (3) in `Möttönen et al. (2004) <https://arxiv.org/abs/quant-ph/0407010>`_.
-
-    Args:
-        row (int): one-based row number
-        col (int): one-based column number
 
     Returns:
-        (float): transformation matrix entry at given row and column
+        np.ndarray[int]: Array of ``2**rank`` integers that make up the Gray code.
     """
-    # (col >> 1) ^ col is the Gray code of col
-    b_and_g = row & ((col >> 1) ^ col)
-    sum_of_ones = 0
-    while b_and_g > 0:
-        if b_and_g & 0b1:
-            sum_of_ones += 1
+    g = np.array([0, 1])
```

Comment:
I was actually just looking at this code and wanting to  make a similar change too 😄 Happy this is being done. Why create strings just to convert them to ints 🤯

Reply:
Haha, funny! :D Yeah, it's really unnecessary!